src/rgw/rgw_rados.cc

```diff
@@ -2423,13 +2423,31 @@ int RGWRados::Bucket::update_bucket_id(const string& new_bucket_id)
  * marker: if filled in, begin the listing with this object.
  * end_marker: if filled in, end the listing with this object.
  * result: the objects are put in here.
- * common_prefixes: if delim is filled in, any matching prefixes are placed here.
- * is_truncated: if number of objects in the bucket is bigger than max, then truncated.
+ * common_prefixes: if delim is filled in, any matching prefixes are
+ * placed here.
+ * is_truncated: if number of objects in the bucket is bigger than
+ * max, then truncated.
  */
-int RGWRados::Bucket::List::list_objects_ordered(int64_t max,
-						 vector<rgw_bucket_dir_entry> *result,
-						 map<string, bool> *common_prefixes,
-						 bool *is_truncated)
+static inline std::string after_delim(std::string_view delim)
+{
+  // assert: ! delim.empty()
+  char e = delim.back();
+  delim.remove_suffix(1);
+  std::string result{delim.data(), delim.length()};
+  if (e < 255) {
```

Comment:
getting a warning here:
```
/home/cbodley/ceph/src/rgw/rgw_rados.cc: In function ‘std::__cxx11::string after_delim(std::string_view)’:
/home/cbodley/ceph/src/rgw/rgw_rados.cc:2438:9: warning: comparison is always true due to limited range of data type [-Wtype-limits]
   if (e < 255) {
       ~~^~~~~
```
it looks like the comparison should be against `(unsigned char)e`

Reply:
sorry;  that works, or uint8_t(e) or something; I'll send a fix when I get to BLR

Reply:
sorry;  that works, or uint8_t(e) or something; I'll send a fix when I get to BLR, if I can't earlier